numpy/core/arrayprint.py

```diff
@@ -493,6 +493,11 @@ def wrapper(self, *args, **kwargs):
 # gracefully handle recursive calls, when object arrays contain themselves
 @_recursive_guard()
 def _array2string(a, options, separator=' ', prefix=""):
+    # Note we use the copied array instead of the input array here (since for
+    # subclasses gathering from the array to format it may be expensive, e.g. if
+    # the buffer is on an accelerator).
+    a = asanyarray(a)
```

Comment:
Is there a reason this is not in the public API call `array2string` instead?

Reply:
I agree with Matti, not sure I see much point in printing lists, etc. with this function (it would not print like a list after all). OTOH, most things in numpy allow array-likes so I do not care much. @tomhennigan I assume you have fixed the issue long ago on the JAX side? Should we just close this?

Reply:
I'm happy to close this out if it's not useful, we have worked around it in JAX. Thanks for the discussion.